pkg/webhook/mutatingwebhook/types/virtualmachine.go

```diff
@@ -0,0 +1,127 @@
+package types
```

Comment:
Wouldn't it be safer to depend on actual CNV API instead of duplicating it here?

Reply:
Yes, we could do that if we're okay with adding the dependency. But when I tried there were a bunch of dependency conflicts so we'd need to resolve those.

Reply:
Having a dependency to CNV API only would be fine. But this API depends on something which creates conflicts on our end then it's not good. OK let's keep it this way. I'm concern with some API changes which would break our webhook though. We won't be able to just update the dependency to verify if it works or not with a newer CNV :(
Can we at least comment it here why we are duplicating their API here?
Also, any idea how we could mitigate risks of breaking API changes?

Reply:
+1 for importing this package if possible, would make maintenance easier (let's imagine if they do some breaking changes and split this package into several packages in the future).

Reply:
Have you tried a generic approach - without introducing the dependency on their library? 
We could either play with the `map[string]interface{}` representation, or use a library to get the value (and check if limits are already set), for example https://github.com/tidwall/gjson should do the job. I know that it's introducing another dependency as well, but it's a more generic one and shouldn't cause any conflict problems with other dependencies.

Another approach (without introducing the dependency on them) would be creating a very simplified representation of what we want to check/get there:
```go

type VM struct {
    Spec struct {
        Template struct {
            Spec struct {
                Domain struct {
                    Resources struct {
                        Limits v1.ResourceList `json:"limits,omitempty"`
                        Requests v1.ResourceList `json:"requests,omitempty"`
                    } `json:"resources,omitempty"`
                } `json:"domain,omitempty"`
            } `json:"spec,omitempty"`
        } `json:"template,omitempty"`
    } `json:"spec,omitempty"`
}
```

Reply:
Used unstructured.Unstructured type for now but will switch to gjson if it's better in later PRs